app/models/company.rb

```diff
@@ -10,8 +12,32 @@ class Company < ApplicationRecord
 
   has_many :business_categories, through: :membership_applications
 
-  has_many :membership_applications
+  has_many :membership_applications, dependent: :destroy, inverse_of: :company
 
   belongs_to :region
 
+
+  def destroy_checks
+
+    error_if_has_accepted_applications?
+
+  end
+
+
+  # do not delete a Company if it has MembershipApplications that are accepted
+  def error_if_has_accepted_applications?
+
+    membership_applications.reload
+    business_categories.reload
```

Comment:
it ensures a fresh read from the db, in case the objects in memory are out of sync with the db.
Ex:  someone has added or changed a membership_application associated with a company